src/RemoteSymbolProvider/StadiaSymbolStoreSymbolProviderTest.cpp

```diff
@@ -0,0 +1,193 @@
+                      const std::vector<SymbolDownloadQuery>& download_queries)
+                      -> Future<ErrorMessageOr<std::vector<SymbolDownloadInfo>>> {
+          // Stadia symbol provider queries only one module each time when making the ggp call.
+          ORBIT_CHECK(download_queries.size() == 1);
+
+          if (ggp_client_state == GgpClientState::kFailing) return {ErrorMessage{"Timeout"}};
+
+          if (download_queries.front().module_name != kOnlyAvailableModuleName ||
+              download_queries.front().build_id != kOnlyAvailableModuleBuildId) {
+            return {ErrorMessage{"Symbol not found"}};
+          }
+
+          SymbolDownloadInfo download_info;
+          download_info.file_id = QString::fromStdString(absl::StrFormat(
+              "symbolFiles/%s/%s", kOnlyAvailableModuleBuildId, kOnlyAvailableModuleName));
+          download_info.url = "valid_url_for_symbol";
+          return {std::vector<SymbolDownloadInfo>{download_info}};
+        });
+  }
+
+  enum class DownloadResultState { kSuccess, kCanceled, kError };
+  void SetUpDownloadManager(DownloadResultState expected_state, std::string error_msg = "") {
+    EXPECT_CALL(download_manager_, Download)
+        .Times(1)
+        .WillOnce([expected_state = std::move(expected_state), error_msg = std::move(error_msg)](
+                      std::string /*url*/, std::filesystem::path /*save_file_path*/,
+                      orbit_base::StopToken /*token*/) -> Future<ErrorMessageOr<CanceledOr<void>>> {
+          switch (expected_state) {
+            case DownloadResultState::kSuccess:
+              return {outcome::success()};
+            case DownloadResultState::kCanceled:
+              return {Canceled{}};
+            case DownloadResultState::kError:
+              return {ErrorMessage{error_msg}};
+          }
+
+          ORBIT_UNREACHABLE();
+        });
+  }
+
+ protected:
+  orbit_symbols::MockSymbolCacheInterface symbol_cache_;
+  orbit_http::MockDownloadManagerInterface download_manager_;
+  orbit_ggp::MockClient ggp_client_;
+  StadiaSymbolStoreSymbolProvider symbol_provider_;
+  std::shared_ptr<orbit_qt_utils::MainThreadExecutorImpl> executor_;
+};
+}  // namespace
+
+TEST_F(StadiaSymbolStoreSymbolProviderTest, RetrieveModuleSuccess) {
+  SetUpGgpClient(GgpClientState::kWorking);
+  SetUpDownloadManager(DownloadResultState::kSuccess);
+
+  orbit_base::StopSource stop_source{};
+  symbol_provider_.RetrieveSymbols(kValidModuleId, stop_source.GetStopToken())
+      .Then(executor_.get(), [](ErrorMessageOr<CanceledOr<std::filesystem::path>> result) {
+        EXPECT_THAT(result, HasNoError());
+        EXPECT_FALSE(IsCanceled(result.value()));
+        const auto& local_file_path = std::get<std::filesystem::path>(result.value());
+        EXPECT_TRUE(
+            local_file_path ==
+            std::filesystem::path(kSymbolCacheDir / "module_path_to_only_available_module_name"));
```

Comment:
This line is in a way testing whether the MockSymbolCache is doing the correct thing, which is not really what should be tested here. I would change this here to 
```
EXPECT_EQ(local_file_path, symbol_cache_.GenerateCachedFileName(kValidModuleId.file_path);
```

Then you could also make the EXPECT_CALL simpler/trivial